### PR TITLE
Add new trigger on error

### PR DESCRIPTION
**Pull request by AnthonySendra**
This PR allows to attach pipes on events `<controller>:error<Action>`, like `document:errorGet`.
The aim is to intercept error and change it to a normal response.

### CODE REVIEW

File: lib/api/controllers/funnelController.js [hunk 1]
```diff
@@ -445,17 +429,16 @@ class FunnelController {
...
     // JSON.stringify(new NativeError()) === '{}', i.e. Error, SyntaxError, TypeError, ReferenceError, etc.
-    this.kuzzle.pluginsManager.trigger('log:error', error instanceof Error && !(error instanceof KuzzleError)
-      ? error.message + '\n' + error.stack
-      : error
+    this.kuzzle.pluginsManager.trigger('log:error', request.error instanceof Error && !(request.error instanceof KuzzleError)
+      ? request.error.message + '\n' + request.error.stack
+      : request.error
     );
 
-    request.setError(error);
```
Comment (AnthonySendra) [inline lib/api/controllers/funnelController.js]: The error is now set directly in `handleProcessRequestError` method

**Comment by scottinet** [comment]
FYI: documentation merged (https://github.com/kuzzleio/documentation/pull/303)

File: lib/api/controllers/funnelController.js [hunk 2]
```diff
@@ -388,26 +388,54 @@ class FunnelController {
...
+              return Bluebird.reject(_error);
+            });
         }
 
-        return Bluebird.reject(_error);
-      })
-      .finally(() => {
-        this.concurrentRequests--;
-        delete this.pendingRequests[request.id];
+        return Bluebird.reject();
```
Comment (scottinet) [inline lib/api/controllers/funnelController.js]: Shouldn't we reject with the `_error` variable?

File: lib/api/controllers/funnelController.js [hunk 3]
```diff
@@ -388,26 +388,54 @@ class FunnelController {
...
+        if (modifiedRequestError.error !== null) {
+          return Bluebird.reject(modifiedRequest.error);
+        }
 
-        if (controllers === this.pluginsControllers && !(error instanceof KuzzleError)) {
-          _error = new PluginImplementationError(error);
+        return modifiedRequestError;
+      })
+      .catch(customError => {
+        if (controllers === this.pluginsControllers && !(customError instanceof KuzzleError)) {
```
Comment (scottinet) [inline lib/api/controllers/funnelController.js]: Should assign `customError` to `_error` first.

**Pull request merged**